pilot/pkg/xds/discovery.go

```diff
@@ -167,7 +167,7 @@ func NewDiscoveryServer(env *model.Environment, instanceID string, clusterID clu
 			debounceMax:       features.DebounceMax,
 			enableEDSDebounce: features.EnableEDSDebounce,
 		},
-		Cache:              model.DisabledCache{},
+		Cache:              env.Cache,
```

Comment:
Are you asking what is the benefit of putting in env? So that we can use it for server and EndpointShards the same thing

Reply:
currently we store in the server and only set to endpointindex

Reply:
Sorry. I am not understanding the problem. The benefit of putting it in Env is we build once based on whether cache is enabled or not and use it every where.

If you do not like this approach, at the minimum, I would start with disabledCache in `EndpointIndex` constructor and change it when setCache is called (similar to how server does - it is initializes with DisabledCache and updated with real cache later) so that we don't check cache != nil every where in EndpointIndex.  #45800 changed at only one place but not sure where else it can break because cache is accessed at multiple places there.